datajoint/declare.py

```diff
@@ -6,14 +6,41 @@
 import pyparsing as pp
 import logging
 
-from .settings import config
 from .errors import DataJointError
 
-STORE_NAME_LENGTH = 8
-STORE_HASH_LENGTH = 43
-HASH_DATA_TYPE = 'char(51)'
 UUID_DATA_TYPE = 'binary(16)'
 MAX_TABLE_NAME_LENGTH = 64
+CONSTANT_LITERALS = {'CURRENT_TIMESTAMP'}  # SQL literals to be used without quotes (case insensitive)
+EXTERNAL_TABLE_ROOT = '~external'
+
+TYPE_PATTERN = {k: re.compile(v, re.I) for k, v in dict(
+    INTEGER=r'(tiny|small|medium|big|)int(\s*\(.+\))?(\s+unsigned)?(\s+auto_increment)?$',
+    DECIMAL=r'(decimal|numeric)(\s*\(.+\))?(\s+unsigned)?$',
+    FLOAT=r'(double|float|real)(\s*\(.+\))?(\s+unsigned)?$',
+    STRING=r'(var)?char\s*\(.+\)$',
+    ENUM=r'enum\s*\(.+\)$',
+    BOOL=r'bool(ean)?$',   # aliased to tinyint(1)
+    TEMPORAL=r'(date|datetime|time|timestamp|year)(\s*\(.+\))?$',
+    INTERNAL_BLOB=r'(tiny|small|medium|long|)blob$',
+    INTERNAL_ATTACH=r'attach$',
+    EXTERNAL_ATTACH=r'attach@(?P<store>[a-z]\w*)$',
+    EXTERNAL_BLOB=r'blob@(?P<store>[a-z]\w*)$',
```

Comment:
I'm totally going to reorder this in the future so that `EXTERNAL_BLOB` comes right after `INTERNAL_BLOB`